frontend/public/components/monitoring/dashboards/index.tsx

```diff
@@ -509,6 +541,25 @@ const Card: React.FC<CardProps> = React.memo(({ panel }) => {
   const ref = React.useRef();
   const [, wasEverVisible] = useIsVisible(ref);
 
+  const [dataSourceInfoLoading, setDataSourceInfoLoading] = React.useState<boolean>(true);
+  const [customDataSource, setCustomDataSource] = React.useState<CustomDataSource>(undefined);
+  const dataSourceName = panel.datasource?.name;
+  const extensions = useExtensions<DataSourceExtension>(isDataSource);
+
+  React.useEffect(() => {
+    if (!dataSourceName) {
+      setDataSourceInfoLoading(false);
+    } else {
+      setDataSourceInfoLoading(true);
+      extensions.forEach(async (extension) => {
```

Comment:
As we are using foreach here we are not waiting for the datasource info to be resolved, so the first time is loaded is falling back to the in-cluster prometheus